ropgadget/gadgets.py

```diff
@@ -72,9 +74,15 @@ def __gadgetsFinding(self, section, gadgets, arch, mode):
                     if not decodes:
                         continue
                     total_size = sum(size for _, size, _, _ in decodes)
-                    expected_size = i * gad_align + gad_size
-                    if total_size != expected_size:
-                        continue
+                    if arch == CS_ARCH_X86:
+                        # x86/x64: only require that bytes decode cleanly and reach the ret
+                        if total_size != (end - start):
+                            continue
+                    else:
+                        # original ROPgadget logic for fixed-width ISAs
+                        expected_size = i * gad_align + gad_size
```

Comment:
I suppose this code should have some different logic now as you changed `start` to `ref - i` instead of  `ref - i * gad_align`.